src/predict_request_validation_utils.cpp

```diff
@@ -483,6 +486,72 @@ Status RequestValidator<RequestType, InputTensorType, IteratorType, ShapeType>::
     return StatusCode::OK;
 }
 
+static size_t getStringInputWidth(const KFSTensorInputProto& src) {
+    size_t maxStringLength = 0;
+    for (const auto& str : src.contents().bytes_contents()) {
+        maxStringLength = std::max(maxStringLength, str.size());
+    }
+    return maxStringLength + 1;
+}
+
+static size_t getStringInputWidth(const tensorflow::TensorProto& src) {
+    size_t maxStringLength = 0;
+    for (const auto& str : src.string_val()) {
+        maxStringLength = std::max(maxStringLength, str.size());
+    }
+    return maxStringLength + 1;
+}
+
+static size_t getStringInputWidth(const InferenceTensor& src) {
+    return 0;
+}
+
+static size_t getStringBatchSize(const KFSTensorInputProto& src) {
+    return src.contents().bytes_contents_size();
+}
+
+static size_t getStringBatchSize(const tensorflow::TensorProto& src) {
+    return src.string_val_size();
+}
+
+static size_t getStringBatchSize(const InferenceTensor& src) {
+    return 0;
+}
+
+template <typename RequestType, typename InputTensorType, typename IteratorType, typename ShapeType>
+Status RequestValidator<RequestType, InputTensorType, IteratorType, ShapeType>::checkStringShapeMismatch(const InputTensorType& proto, const ovms::TensorInfo& inputInfo, Status& finalStatus, Mode batchingMode, Mode shapeMode) const {
+    const auto& shape = inputInfo.getShape();
+    bool mismatch = false;
+    size_t width = getStringInputWidth(proto);
+    size_t batchSize = getStringBatchSize(proto);
+    if (batchingMode == AUTO) {  // Skip batch dimension
+        if (!shape[1].match(static_cast<dimension_value_t>(width))) {
+            mismatch = true;
+        }
+    } else {  // Do not skip batch dimension
+        if ((!shape[0].match(static_cast<dimension_value_t>(batchSize))) || (!shape[1].match(static_cast<dimension_value_t>(width)))) {
+            mismatch = true;
+        }
+    }
+    if (!mismatch) {
+        return StatusCode::OK;
```

Comment:
This will not work and simple shape=auto + string tests should show that.
Scenario: shape=auto, model shape [2, -1], request string with shape [3] ["A","B","C"]

`mismatch` will be false, and we will return StatusCode::OK instead of BATCHSIZE CHANGE REQUIRED